### PR TITLE
Persisted State

### PR DESCRIPTION
**Pull request by ErikSin**
This PR just encapsulates the creation of a `usePersistedState`. I will put the rest of the zustand store into another PR.

The motivation is to use a storage system that can be encrypted (for things like passcode...etc). Mapeo Mobile uses Async-storage, which cannot be encrypted. This uses mmkv. 

With this use hook, I tried to follow the [architecture used in Mapeo Mobile](https://github.com/digidem/mapeo-mobile/blob/develop/src/frontend/hooks/usePersistedState.ts). It is a function that returns a hook.

Two major difference. This hook uses mmkv, which is synchronous, meaning there is no [`status`](https://github.com/digidem/mapeo-mobile/blob/ca167c257b90ba74b0e63e623f405aed98d86145/src/frontend/hooks/usePersistedState.ts#L23). This means we will have to take this out when moving things over

And to maintain typesafety, I created a custom type, which should hold all the keys for persisted state

### CODE REVIEW

**Review by achou11 — COMMENTED** [review]
Curious about why this needs to be handled with Zustand when `react-native-mmkv` provides hooks for interfacing with the storage instances that are created?

https://github.com/mrousavy/react-native-mmkv/blob/HEAD/docs/HOOKS.md

Also, added a commit to run prettier on the file because I realized that I haven't set anything up for the repo. Created a ticket to get that done in #24

**Review by achou11 — APPROVED** [review]
Tentatively approving assuming no major concerns come up while re-reviewing. Think it would be nice to have an example implemented in the app

Potentially out of scope for this, but another thing that we'll have to account for is storing the encryption key itself, which shouldn't be done in mmkv. We should use something that uses the Keychain (iOS) and Keystore (Android) APIs, which `react-native-keychain` does (albeit with a somewhat weirdly-named interface). This may be relevant for this PR (or as a subsequent one) because the flow would be something like this:

1. On first app startup, create encryption key and store via react-native-keychain
2. Use the encryption key to create the MMKV instance (specifying the `encryptionKey` opt)
3. Create the persistedState hook based on the encrypted MMKV instance's id

Don't have a full idea of how to go about that, but food for thought!

Relevant links:

- React Native docs around security: https://reactnative.dev/docs/security#storing-sensitive-info
- React Native Keychain: https://github.com/oblador/react-native-keychain
- Implementation suggestion for using React Native Keychain with a generic key-value interface: https://github.com/oblador/react-native-keychain/issues/291#issuecomment-682460091

**Comment by ErikSin** [comment]
> Also, added a commit to run prettier on the file because I realized that I haven't set anything up for the repo. Created a ticket to get that done in #24

Good point, Im going to set that up first before we merge anything

**Comment by ErikSin** [comment]
> Curious about why this needs to be handled with Zustand when `react-native-mmkv` provides hooks for interfacing with the storage instances that are created?

TBH my intention was to just set up a zustand store, and while I was reading the documentation I saw that there were easy middleware to persist state. I kind of created this on autpilot, and hadn't really considered just using the hooks....

that being said, I agree that we should incorporate migration (especially for things like `settings`), so I think using zustand does make sense!

**Comment by ErikSin** [comment]
Agreed that the encryption key is a little out of scope as I think require a little more research. There are a few sensitive things that we are going to start storing, so I will create a ticket to look into that work

File: src/frontend/hooks/usePersistedState.ts
Comment (ErikSin) [reply src/frontend/hooks/usePersistedState.ts]: Im create another ticket for version/migrating. I started doing some of that work, but its a little bigger than I thought so worth putting some time into it

File: src/frontend/hooks/usePersistedState.ts
Comment (ErikSin) [reply src/frontend/hooks/usePersistedState.ts:16]: No, it is handled by zustand!

**Pull request merged**
